design-proposals/migration-target.md

```diff
@@ -0,0 +1,256 @@
+This proposal is part of a larger scheduling enhancement picture.
+Cluster admins are also looking for, for instance [descheduler integration](https://github.com/kubevirt/community/pull/2580), [load aware scheduling](https://github.com/kubevirt/user-guide/pull/621).
+This proposal is just an additional piece of the puzzle but, again, is the tool for exceptions and the corner cases, not the norm.
+
+Such a capability is expected from traditional virtualization solutions but, with certain limitations, is still pretty common across the most popular cloud providers (at least when using dedicated and not shared nodes).
+- For instance on Amazon EC2 the user can already live-migrate a `Dedicated Instance` from a `Dedicated Host` to another `Dedicated Host` explicitly choosing it from the EC2 console, see: https://repost.aws/knowledge-center/migrate-dedicated-different-host
+- also on Google Cloud Platform Compute Engine the user can easily and directly live-migrate a VM from a `sole-tenancy` node to another one via CLI or REST API, see: https://cloud.google.com/compute/docs/nodes/manually-live-migrate#gcloud
+- Project Harvester, an HCI solution built on top of Kubevirt is also offering the capability of live migrating a VM to a named node, see: https://docs.harvesterhci.io/v1.3/vm/live-migration/#starting-a-migration
+  Project Harvester approach is although different from this proposal, its implications are analyzed in section [Project Harvester approach](#project-harvester-approach)
+
+On the technical side something like this can already be indirectly achieved playing with node labels and affinity but nodeSelector and affinity are going to be defined as VM properties that are going to stay while here we are focusing just on setting the desired target of a one-off migration attempt without any future side effect on the VM.
+The motivation is to better define a boundary between what is an absolute and long-lasting property of a VM (like affinity) with what is just an optional property of the single migration attempt.
+This could also be relevant in terms of personas: we could have the VM owner/developer that is going to specify long-lasting affinity for a VM that is part of an application composed by different VMs and pods and a cluster admin/operator that needs to temporary override that for maintenance reasons.
+On the other side the VM owner is not required/supposed to be aware of node names. 
+
+## Goals
+- A user allowed to trigger a live-migration of a VM and list the nodes in the cluster is able to rely on a simple and direct API to try to live migrate a VM to a specific node (or a node within a set of nodes identified by adding node affinity constraints).
+- The live migration then can successfully complete or fail for various reasons exactly as it can succeed of fail today for other reasons.
```

Comment:
This is stating that in term of exit status, a migration to selected node via the optional API should behave exactly as migration behaves today when the target node is selected by the scheduler: no side effects or try this first and then eventually let the scheduler decide or anything like that.

Reply:
Today live migrations can fail or many reasons.
With this change there is no change to this, live migrations can still fail in many ways.

Technically it's really no change at all, because live migration depens on meeting scheduling constraints. constraints are cpu flags, memory, storage etc - and with this feature locality will be simply one additional constratint,. but the failure more is still the same.

Reply:
IMO this should be dropped as nothing special is added or changed.

Reply:
done, dropped